core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java

```diff
@@ -72,15 +75,17 @@ public LocalInputSource(
     this.baseDir = baseDir;
     this.filter = baseDir != null ? Preconditions.checkNotNull(filter, "filter") : filter;
     this.files = files == null ? Collections.emptySet() : files;
-
     if (baseDir == null && CollectionUtils.isNullOrEmpty(files)) {
       throw new IAE("At least one of baseDir or files should be specified");
     }
+    securityConfig.validateFileAccess(baseDir);
+    securityConfig.validateFileAccess(files);
```

Comment:
Maybe out of this PR's scope but it could be useful to support pattern based validation for `filter`.